--- /usr/bin/perldoc	2003-06-02 16:49:30.000000000 +0300
+++ /usr/bin/perldoc.patched	2003-07-23 15:58:40.000000000 +0300
@@ -43,7 +43,8 @@
 	my $me = $0;		# Editing $0 is unportable
 	$me =~ s,.*/,,;
 	die <<EOF;
-Usage: $me [-h] [-r] [-i] [-v] [-t] [-u] [-m] [-n program] [-l] [-F] [-X] PageName|ModuleName|ProgramName
+Usage: $me [-h] [-r] [-i] [-v] [-t] [-u] [-m] [-n program] [-L language] 
+           [-l] [-F] [-X] PageName|ModuleName|ProgramName
        $me -f PerlFunc
        $me -q FAQKeywords
 
@@ -81,6 +82,7 @@
     -n   Specify replacement for nroff
     -l   Display the module's file name
     -F   Arguments are file names, not modules
+    -L   Choose documentation translation language (if any)
     -v	 Verbosely describe what's going on
     -X	 use index if present (looks for pod.idx at $Config{archlib})
     -q   Search the text of questions (not answers) in perlfaq[1-9]
@@ -112,9 +114,9 @@
     unshift(@ARGV, Text::ParseWords::shellwords($ENV{"PERLDOC"}));
 }
 
-use vars qw( $opt_m $opt_h $opt_t $opt_l $opt_u $opt_v $opt_r $opt_i $opt_F $opt_f $opt_X $opt_q $opt_n $opt_U );
+use vars qw( $opt_m $opt_h $opt_t $opt_l $opt_u $opt_v $opt_r $opt_i $opt_F $opt_f $opt_X $opt_q $opt_n $opt_U $opt_L );
 
-getopts("mhtluvriFf:Xq:n:U") || usage;
+getopts("mhtluvriFf:Xq:n:UL:") || usage;
 
 usage if $opt_h;
 
@@ -184,6 +186,9 @@
 elsif ($opt_q) {
     @pages = ("perlfaq1" .. "perlfaq9");
 }
+elsif ($opt_L) {
+    @pages = map { 'POD2::' . uc($opt_L) . '::' . $_ } @ARGV;
+}
 else {
     @pages = @ARGV;
 }
